File: LeetCode/[E] - 20. Valid Parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        m = {"(": ")", "{": "}", "[": "]"}
        stack = []
        for b in s:
            if b in m:
                stack.append(b)
            elif stack and stack[-1] in m and m[stack[-1]] == b:
                stack.pop()
            else:
                return False
        return len(stack) == 0

File: LeetCode/[E] - 20. Valid Parentheses/solution.go
package main

func isValid(s string) bool {
	m := map[rune]rune{'(': ')', '{': '}', '[': ']'}
	stack := make([]rune, 0)
	for _, r := range s {
		if _, ok := m[r]; ok {
			stack = append(stack, r)
		} else if len(stack) > 0 && m[stack[len(stack)-1]] != 0 && m[stack[len(stack)-1]] == r {
			stack = stack[:len(stack)-1]
		} else {
			return false
		}
	}
	return len(stack) == 0
}
